src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_NativeFrame.cs

```diff
@@ -25,49 +25,89 @@ public async Task When_NavigateForward()
 			var style = Windows.UI.Xaml.Application.Current.Resources["NativeDefaultFrame"] as Style;
 			Assert.IsNotNull(style);
 
-			var SUT = new Frame() {
+			var SUT = new Frame()
+			{
 				Style = style
 			};
 
 			TestServices.WindowHelper.WindowContent = SUT;
 
-			int GetAllMyPages()
-				=> SUT.EnumerateAllChildren(v => v is MyPage).Count();
+			await SUT.WaitForPages(0);
+
+			SUT.Navigate(typeof(MyPage));
+
+			await SUT.WaitForPages(1);
+
+			SUT.Navigate(typeof(MyPage));
+
+			await SUT.WaitForPages(2);
 
-			/// Actively waiting for pages to be stacked is
-			/// required as NativeFramePresenter.UpdateStack awaits
-			/// for animations to finish, and there's no way to determine
-			/// from the Frame PoV that the animation is finished.
-			async Task WaitForPages(int count)
+			SUT.GoBack();
+
+			await SUT.WaitForPages(1);
+		}
+
+
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_NavigateBackSkipingPages()
+		{
+			var style = Windows.UI.Xaml.Application.Current.Resources["NativeDefaultFrame"] as Style;
+			Assert.IsNotNull(style);
+
+			var SUT = new Frame()
 			{
-				var sw = Stopwatch.StartNew();
+				Style = style
+			};
 
-				while(sw.Elapsed < TimeSpan.FromSeconds(5))
-				{
-					await TestServices.WindowHelper.WaitForIdle();
+			TestServices.WindowHelper.WindowContent = SUT;
 
-					if (GetAllMyPages() == count)
-					{
-						break;
-					}
-				}
+			await SUT.WaitForPages(0);
 
-				Assert.AreEqual(count, GetAllMyPages());
-			}
+			SUT.Navigate(typeof(MyPage));
 
-			await WaitForPages(0);
+			await SUT.WaitForPages(1);
 
 			SUT.Navigate(typeof(MyPage));
 
-			await WaitForPages(1);
+			await SUT.WaitForPages(2);
 
 			SUT.Navigate(typeof(MyPage));
 
-			await WaitForPages(2);
+			await SUT.WaitForPages(3);
+
+			//Remove the previous page to jump from page 3 to page 1
+			SUT.BackStack.Remove(SUT.BackStack.LastOrDefault());
 
 			SUT.GoBack();
 
-			await WaitForPages(1);
+			await SUT.WaitForPages(1);
+		}		
+	}
+
+	internal static class FrameExtensions
```

Comment:
Put this in its own file here https://github.com/unoplatform/uno/tree/master/src/Uno.UI.RuntimeTests/Extensions